src/client/e2e/blotter.spec.ts

```diff
@@ -0,0 +1,177 @@
+import { expect, Page } from "@playwright/test"
+import { test } from "./fixtures"
+// import { fxColDef, fxColFields } from "../src/App/Trades/TradesState/colConfig"
+// import { FxTrade } from "../src/services/trades"
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fs = require("fs")
+
+const getTradeColIndex = () => {
+  // const tradeIndex = fxColFields.indexOf(
+  //   fxColDef.tradeId as unknown as keyof FxTrade,
+  // )
+  // // + 1 to account for status indicator
+  // return tradeIndex + 1
+  return 1
+}
+
+test.describe("Trade Blotter", () => {
+  let tilePage: Page
+  let blotterPage: Page
+
+  test.beforeAll(async ({ context, fxOpenfinPagesRec }, testInfo) => {
+    if (testInfo.project.name === "openfin") {
+      const mainWindow = fxOpenfinPagesRec["mainWindow"]
+      await mainWindow.evaluate(async () => {
+        const currentWindow = window.fin.desktop.Window.getCurrent()
+        currentWindow.maximize()
+        return window.fin
+      })
+
+      tilePage = fxOpenfinPagesRec["fx-tiles"]
+      blotterPage = fxOpenfinPagesRec["fx-blotter"]
+    } else {
+      const pages = await context.pages()
+      const mainWindow = pages.length > 0 ? pages[0] : await context.newPage()
+
+      await mainWindow.goto(`${process.env.URL_PATH}`)
+
+      tilePage = mainWindow
+      blotterPage = mainWindow
+    }
+  })
+  test("When user hovers over a row on the Blotter, it should highlight that row", async () => {
+    const firstRow = await blotterPage.locator('[role="grid"] > tr').nth(0)
+    const color = await firstRow.evaluate((element) =>
+      window.getComputedStyle(element).getPropertyValue("background-color"),
+    )
+    await firstRow.hover()
+    await expect(firstRow).not.toHaveCSS("background-color", color)
+  })
+  test("when user buys a currency, the new row should flash briefly", async () => {
+    const tile = await tilePage.locator('[data-testid="Buy-EURUSD"]').nth(0)
+    await tile.click()
+
+    const tradeId = await tilePage
+      .locator("[data-testid='trade-id']")
+      .innerText()
+    const newRow = await blotterPage.getByTestId(`trades-grid-row-${tradeId}`)
+    await expect(newRow).toHaveCSS("animation", /1s ease-in-out/)
+  })
+  test("when user clicks on the header of any column, it should sort it (depending on number of clicks, can be ascending or descending)", async () => {
+    const firstTradeID = await blotterPage
```

Comment:
just for a little more clarity, call it `firstRowTradeID`

Reply:
```
await blotterPage.locator('[role="grid"] > tr')
  .nth(0)
  .locator('td')
  .nth(getTradeIdColIndex())
  .textContent()
```

This makes explicit that you are first selecting the first row, then the trade id cell in it.